enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,301 @@
+- [Ingress API](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#ingress-v1-networking-k8s-io). Implemented in Kubernetes.
+
+However the implementation of these APIs (or parts of them) depend on the ingress operator:
+- The ingress operator creates the default ingress controller to manage the backends for the routes. See [the drawbacks](#drawbacks) for further implications.
+  - The ingress objects are implicitly managed by the default ingress controller. See [the ingress to route controller risk](#ingress-to-route-controller) for further implications.
+- For [the component routes](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html#spec-componentroutes) the ingress operator creates the required role and role bindings to get the custom TLS certificate secret from `openshift-config` namespace. See [the component routes risk](#component-routes) for further implications.
+
+#### Console operator
+
+To prevent the console operator from going degraded, [the console route check](https://github.com/openshift/console-operator/blob/master/pkg/console/controllers/healthcheck/controller.go#L122) should be skipped if the ingress capability is not enabled in `ClusterVersion`.
+
+#### Authentication operator
+
+No change to the authentication operator is needed until the implementation of the ingress capability for the standalone OpenShift.
+HyperShift doesn't use the authentication operator, the authentication server is [managed by HyperShift's control plane directly](https://github.com/openshift/hypershift/blob/7cb87788779ae928d90d0e7760c1e9359f04e58a/control-plane-operator/controllers/hostedcontrolplane/oauth/deployment.go#L123-L127).
+
+#### Monitoring operator
+
+No change to the monitoring operator is needed until the implementation of the ingress capability for the standalone OpenShift.
+The monitoring operator skips the alert manager's route check if `ingress` cluster operator is not there.
+
+### Risks and Mitigations
+
+#### Partially implemented capability
+
+The ingress capability will become a part of the capability API. Meanwhile its implementation will be incomplete due to the missing support of the standalone OpenShift. However the usage of the capability is an important stepping stone, it unifies the implementation of the optional ingress for all the OpenShift installation types.
+
+#### Component routes
+
+In the absence of the ingress operator, the component routes cannot be customized by the owner components due to the lack of the RBAC.
+The cluster administrator can customize a component route with a custom TLS certificate secret but the component owner won't be able to access the secret. This may be misleading for the cluster administrators. However the component routes were never fully implemented on HyperShift (authentication operator is not used).
```

Comment:
what is a `component route` compared to a `route`?  (since the `route` api will be entirely non-functional on hypershift when the ingress capability is disabled, i'm confused about what this paragraph is saying)

Reply:
[Component Routes field from Ingress Config API](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html#spec-componentroutes):
> componentRoutes is an optional list of routes that are managed by OpenShift components that a cluster-admin is able to configure the hostname and serving certificate for. The namespace and name of each route in this list should match an existing entry in the status.componentRoutes list. To determine the set of configurable Routes, look at namespace and name of entries in the .status.componentRoutes list, where participating operators write the status of configurable routes.

They are used to customize the [web console](https://docs.openshift.com/container-platform/4.13/web_console/customizing-the-web-console.html#customizing-the-console-route_customizing-web-console) and [authentication](https://docs.openshift.com/container-platform/4.13/authentication/configuring-internal-oauth.html#customizing-the-oauth-server-url_configuring-internal-oauth) routes. 

I updated this chapter to better detail the component routes.

Reply:
thanks....i assume the componentRoutes are implemented as Ingress resources(backed by any ingress controller on the cluster), not directly as Routes?  otherwise componentRoutes would be totally broken just by virtue of the route api being non-functional when ingress-operator is disabled.

Reply:
Well, yeah, I put the emphasis on the minor detail which will be broken. But so far all the component routes have been implemented as routes. So, indeed, I can shorten this chapter to say just this.

Reply:
The chapter [updated](https://github.com/openshift/enhancements/commit/78a5a1083bdc32f62f0c2ef23ac8c9c77f538d62) to mention the absence of the route implementation.

Reply:
thanks, resolving this one